qemu/exec.c

```diff
@@ -1076,18 +1074,26 @@ static void ram_block_add(struct uc_struct *uc, RAMBlock *new_block)
      * QLIST (which has an RCU-friendly variant) does not have insertion at
      * tail, so save the last element in last_block.
      */
-    RAMBLOCK_FOREACH(block) {
-        last_block = block;
-        if (block->max_length < new_block->max_length) {
-            break;
+    if (uc->ram_list.freed || new_block->max_length > uc->target_page_size) {
```

Comment:
I fail to understand the reason of `new_block->max_length > uc->target_page_size` check. Why in such case you can't append ram blocks?

Reply:
The `ram_list` is ordered by the size of the ram block, so adding a block with `new_block->max_length > uc->target_page_size` would not necessary added at the end of the list. A future optimization could add an comparison of the max_length of the last block in the list and the new block.

Reply:
Ohh, I see. Thanks for the explanation.